tests/utils/btc.test.ts

```diff
@@ -0,0 +1,49 @@
+import { isTaproot } from '../../src/utils/btc';
+import { networks } from 'bitcoinjs-lib';
+
+const mainnetTaprootAddress = 'bc1p8wsazny8z6l8jv9whagu6zrx43t2lxu9q7xltlp3w44qjja9t3hscc85ux';
+const signetTaprootAddress = 'tb1p2wl2dglg0sqv4r8l7r4uc5av72hyty8zprelfa4kwxw9xhqkv55s3kz7ze';
+
+describe('isTaproot', () => {
+  it('should return true for a valid mainnet Taproot address', () => {
+    expect(isTaproot(mainnetTaprootAddress, networks.bitcoin)).toBe(true);
+  });
+
+  it('should return true for a valid signet Taproot address', () => {
+    expect(isTaproot(signetTaprootAddress, networks.testnet)).toBe(true);
+  });
+
+  it('should return false for a mainnet non-Taproot address', () => {
+    const nativeSegWidthAddress = 'bc1qem92n3xk2rm72mua7jq66m700m3r2ama60mc35';
```

Comment:
Other addresses we define at the top, while here we define them inside the test. Also, this test should be straightforward to randomize